events/events.go

```diff
@@ -8,48 +8,74 @@
 // by the Apache License, Version 2.0, included in the file
 // licenses/APL.txt.
 
-/*
-Package events provides the internal event system.
-*/
 package events
 
-type Subscription[T any] chan T
+import (
+	"github.com/ipfs/go-cid"
+	"github.com/libp2p/go-libp2p/core/event"
+	"github.com/libp2p/go-libp2p/core/peer"
+)
 
-// Channel represents a subscribable type that will expose inputted items to subscribers.
-type Channel[T any] interface {
-	// Subscribe subscribes to the Channel, returning a channel by which events can
-	// be read from, or an error should one occur (e.g. if this object is closed).
-	//
-	// This function is non-blocking unless the subscription-buffer is full.
-	Subscribe() (Subscription[T], error)
+const (
+	// WildCardEventName is the alias used to subscribe to all events.
+	WildCardEventName = "*"
+	// MergeCompleteEventName is the name of the database merge complete event.
+	MergeCompleteEventName = "db:merge-complete"
+	// UpdateEventName is the name of the database update event.
+	UpdateEventName = "db:update"
+	// ResultsEventName is the name of the database results event.
+	ResultsEventName = "db:results"
+	// MergeRequestEventName is the name of the net merge request event.
+	MergeRequestEventName = "net:merge"
+	// PubSubEventName is the name of the network pubsub event.
+	PubSubEventName = "net:pubsub"
+	// ConnectEventName is the name of the network connect event.
+	ConnectEventName = "net:connect"
+)
```

Comment:
I'd like to keep the event names a human readable string for debugging purposes.

Having the events in one package does have the issue you mentioned, but it also makes it much easier to know what events can be subscribed to. There's also the issue of importing an entire package just for an event type.

Reply:
but the package that wants to listed to a specific event usually depends on the package that fires that event as it needs to receive the package-event-specific object.

> I'd like to keep the event names a human readable string for debugging purposes

You can declare string enum:
```go
type EventName string
```
and in `db` package:
```go
const (
    // MergeCompleteEventName is the name of the database merge complete event.
    MergeCompleteEventName = events.EventName("db:merge-complete")
    // UpdateEventName is the name of the database update event.
    UpdateEventName = events.EventName("db:update")
    // ResultsEventName is the name of the database results event.
    ResultsEventName = events.EventName("db:results")
)
```

Reply:
> but the package that wants to listed to a specific event usually depends on the package that fires that event as it needs to receive the package-event-specific object.

This won't always be the case, and will cause problems since the `db` package is internal.

> You can declare string enum:

I've added an `event.Name` type.